pkg/endpointmanager/cell.go

```diff
@@ -221,6 +221,7 @@ func newDefaultEndpointManager(p endpointManagerParams) endpointManagerOut {
 			},
 			OnStop: func(cell.HookContext) error {
 				cancel()
+				mgr.OnShutdown()
```

Comment:
Do we really only want to do this `if EndpointGCInterval > 0`? I guess that's just mostly always true?